python_modules/dagster/dagster_tests/definitions_tests/test_dynamic_partitions.py

```diff
@@ -204,3 +210,80 @@ def test_has_partition_key():
         assert partitions_def.has_partition_key("apple", dynamic_partitions_store=instance)
         assert partitions_def.has_partition_key("banana", dynamic_partitions_store=instance)
         assert not partitions_def.has_partition_key("peach", dynamic_partitions_store=instance)
+
+
+def test_dynamic_outputs_dynamic_partitions():
+    """Test the interoperability of dagster's dynamic output system with the dynamic partitioning system."""
+    partitions = ["apple", "banana", "cantaloupe"]
+    values_per_partition = {
+        "apple": 1,
+        "banana": 2,
+        "cantaloupe": 3,
+    }
+
+    partitions_def = DynamicPartitionsDefinition(name="fruits")
+
+    @dg.op(out=dg.DynamicOut())
+    def fan_out(context: dg.OpExecutionContext):
+        for partition_key in context.partition_keys:
+            yield dg.DynamicOutput(values_per_partition[partition_key], mapping_key=partition_key)
+
+    @dg.op
+    def process_fruit_val(context, val: int):
+        return val + 1
+
+    @dg.op(out=dg.Out(dagster_type=dg.Nothing))
+    def collect_fruit_vals(context, vals):
+        assert vals == [2, 3, 4]
+
+    @dg.graph_asset(partitions_def=partitions_def, backfill_policy=dg.BackfillPolicy.single_run())
+    def doubly_dynamic_asset():
+        vals = fan_out().map(process_fruit_val)
+        return collect_fruit_vals(vals.collect())
+
+    with instance_for_test() as instance:
+        instance.add_dynamic_partitions(dg._check.not_none(partitions_def.name), partitions)  # noqa: SLF001
+
+        result = materialize(
+            [doubly_dynamic_asset],
+            instance=instance,
+            tags={
+                ASSET_PARTITION_RANGE_START_TAG: "apple",
+                ASSET_PARTITION_RANGE_END_TAG: "cantaloupe",
+            },
+        )
+
+        assert result.success
+        assert result._output_capture == {  # noqa: SLF001
+            StepOutputHandle(
```

Comment:
theres gotta be a better way to express this without using `_output_capture` or StepOutputHandle

Reply:
Yea I can do it with output_for_node I think